File: safe-online-sdk/src/main/java/net/link/safeonline/sdk/ws/ping/PingClientImpl.java
/*
 * SafeOnline project.
 *
 * Copyright 2006-2007 Lin.k N.V. All rights reserved.
 * Lin.k N.V. proprietary/confidential. Use is subject to license terms.
 */

package net.link.safeonline.sdk.ws.ping;

import java.net.ConnectException;

import javax.xml.ws.BindingProvider;

import net.lin_k.safe_online.ping.PingPort;
import net.lin_k.safe_online.ping.PingService;
import net.lin_k.safe_online.ping.Request;
import net.link.safeonline.ping.ws.PingServiceFactory;
import net.link.safeonline.sdk.ws.AbstractMessageAccessor;

import com.sun.xml.ws.client.ClientTransportException;


public class PingClientImpl extends AbstractMessageAccessor implements PingClient {

    private final PingPort pingPort;


    /**
     * Main constructor.
     *
     * @param location
     */
    public PingClientImpl(String location) {

        PingService pingService = PingServiceFactory.newInstance();
        this.pingPort = pingService.getPingPort();

        BindingProvider bindingProvider = (BindingProvider) this.pingPort;
        bindingProvider.getRequestContext().put(BindingProvider.ENDPOINT_ADDRESS_PROPERTY,
                location + "/safe-online-ws/ping");
    }

    public void ping() throws ConnectException {

        Request request = new Request();
        try {
            this.pingPort.pingOperation(request);
        } catch (ClientTransportException e) {
            throw new ConnectException(e.getMessage());
        } catch (Exception e) {
            throw retrieveHeadersFromException(e);
        } finally {
            retrieveHeadersFromPort(this.pingPort);
        }
    }
}

File: safe-online-sdk-dotnet/src/main/cs/ws/ping/PingClientImpl.cs
/*
 * SafeOnline project.
 * 
 * Copyright 2006-2008 	Lin.k N.V. All rights reserved.
 * Lin.k N.V. proprietary/confidential. Use is subject to license terms.
 */

using System;
using System.Collections.Generic;
using System.ServiceModel;
using System.Diagnostics;
using PingWSNamespace;

namespace safe_online_sdk_dotnet
{
	/// <summary>
	/// Client implementation of the OLAS Ping Web Service.
	/// 
	/// This is only a proof-of-concept implementation.
	/// </summary>
	public class PingClientImpl : PingClient
	{
		private static TraceSource traceSource = new TraceSource("TraceTest");
		
		private PingPortClient pingPortClient;
		
		public PingClientImpl(string location) {
			traceSource.TraceInformation("ping client constructor");
			BasicHttpBinding binding = new BasicHttpBinding();
			EndpointAddress address = 
               new EndpointAddress("http://" + location + "/safe-online-ws/ping");
			this.pingPortClient = new PingPortClient(binding, address);
		}
		
		public void ping() {
			Request request = new Request();
			this.pingPortClient.PingOperation(request);
		}
	}
}
